tools/update-prod-static

```diff
@@ -34,6 +34,9 @@ os.chdir(settings.DEPLOY_ROOT)
 os.makedirs("var/log", exist_ok=True)
 fp = open('var/log/update-prod-static.log', 'w')
 
+# Setup zulip git version cache file
+run(['./tools/cache-zulip-git-version'], stdout=fp, stderr=fp)
```

Comment:
I think the Travis CI error is because the way we build release tarballs is I think in a context where we don't have a Git repository by by the time we're running `update-prod-static` (we first make a fresh checkout with `git archive` in that toolchain to avoid issues with untracked files, etc., in the development environment).  So I think perhaps we want to run `./tools/cache-zulip-git-version` in `tools/build-release-tarball` and in `tools/upgrade-zulip-from-git` instead, as the two code paths that call into `tools/update-prod-static`?  I think that'd also avoid confusing results like the value here being cached in a development environment if we ever make `tools/update-prod-static` directly runnable there for debugging.

Reply:
I was actually writing the version file to the `var/` directory, assuming we don't need this in a `release-tarball`. Do we need this information when running from a release tarball?

Reply:
@timabbott Also, do we need this information in a release tarball? Just having the version set when creating the tarball would be good enough, right? If not, what would be a good place to drop the cache file for the version info? Right now I put it in `var/` assuming we only need it when running from `git`.

Reply:
We definitely want it in release tarballs.

I think `var/` is a fine place to put it if that directory ends up being included in a release tarball.  If it doesn't, which seems likely, we could just put it in the root of the repository; I think we have similar other files like `version` in the root of release tarballs.